### ISSUE TITLE
No mention in documentation that minetest.settings:get() returns nil if the default value was not changed

### ISSUE THREAD

**Issue by loffren174** [minetest/minetest#11736]
##### Minetest version
Minetest 5.4.1 (Linux)

##### Summary
the lua documentation refers to the builtin settingtypes.txt for documentation on how it works.
Additionally, it lists the methods of 'Settings': minetest.settings:get(),minetest.settings:get_bool(),etc.
What is NOT told in either of these is that if the user does not change the default, this simply returns nil, and NOT the default shown in settingtypes.txt. This is completely counterintuitive considering the explanation in builtin's settingtypes.txt that 'string values with no default specified will be the empty string' - well, they aren't, or rather, they aren't unless your code deliberately overrides the values to set the empty string, otherwise you won't ever know, because only nil is returned, or any value you change it to, but not the default value.

##### Details

So, the following code will not work:
local mymod_default_value=minetest.settings:get_bool("mymod_default_value")
it will return 'nil' even if you set a default in settingtypes.txt, such as:

```
# This is a value that does things.
mymod_default_value (mymod default value that does things) string some_string

# This is a boolean that does things.
mymod_default_flag (default_flag) bool true
```

This is even more hazardous if get_bool is used:
```
local mymod_default_flag=minetest.settings:get_bool("mymod_default_flag")
```
which would counterintuitively set the flag to 'nil', if the default 'true' has not been changed. 

But instead of simply returning the actual default value, minetest.settings:get_bool lets you specify a default value to return if not found - no doubt because this required 'get(key) OR value' pattern inevitably fails with 'false'. I'm sure it makes the code tidier to have the defaults as fallback entirely outside settingtypes and not guaranteed to be fetched from there, but if this is required from the modder, at least this should be mentioned in the documentation of the 'minetest.settings' methods that you cannot effectively query for the defaults set in settingtypes.txt but have to also have them set in the code as well, to avoid people chasing mysterious nil values.

(Maybe I was the only one until now, but this wasted a whole afternoon on finding the reason for these nil-queries despite constantly looking at how other mods use settingtypes and ruling out invalid syntax/formatting/OS-issues with linebreaks and UTF-8, until I realized the issue is actually in the use of the minetest.settings functions.)

##### Steps to reproduce

settingtypes.txt:
```
# This is a string that, as claimed in builtin's settingtypes.txt, would be set to the empty string. 
# Noone will ever know, because minetest.settings:get(my_default_string) will return nil, and not the empty string, unless it is changed to be different from the default.
mymod_default_string (default_string) string

# This is a boolean that does things.
mymod_default_flag (default_flag) bool true
```
init.lua:
```
local mymod_default_string=minetest.settings:get("mymod_default_string") -- will actually set the value to nil, not empty string, if user didn't change it
local mymod_default_flag=minetest.settings:get_bool("mymod_default_flag") -- will actually set the value to nil if user didn't change it
print(mymod_default_string) -- prints nil, and not the empty string as claimed in settingtypes.txt
print(mymod_default_flag) -- prints nil, and not the default value 'true' in settingtypes.txt
```

Edited for correct syntax of the print lines, and added an attached demonstration mod. You can observe it having the values 'empty string, true', but the print resulting in 'nil,nil'.
[settingstestmod.zip](https://github.com/minetest/minetest/files/7429664/settingstestmod.zip)

**Comment by sfan5** [comment]
> I provided a PR regarding option 2, but I'd be comfortable with whatever solution that provides this information in a reasonable location for people trying to work with settingtypes.txt.

The documentation should clarify this yes (option 2). Though note that as SmallJoker said `settingstypes.txt` only exists for UI reasons.

**Comment by loffren174** [comment]
> **settingstypes.txt has no functional connection**

I see, thanks for clarifying this! I might have overlooked something, but I don't recall finding this anywhere in the lua doc (although I'm not sure it would entirely fit there- but where else would people look if they want to make their own settingtypes.txt file? I searched for mentions of 'settings' in the lua doc, and didn't find this.) Now it makes sense to me why the 'nil' returns happen and why the comments in settingtypes are written the way they are.  Also, the settings hierarchy was something I was wondering about before already in another context. 

Maybe the best way to convey this would be still in the 'settings' section, to mention this lookup hierarchy of the settings, and that settingtypes.txt only serves as a UI template to provide custom values, which are then returned by the settings.get()/get_bool() functions. I'm tempted to make a PR simply adding your explanation there, but seeing as there might be more things I don't know and would get wrong, I won't try to make a 'half-correct' addition to the docs which ends up being just as misleading, until I spent more time working with the settings hierarchies.

> The problem you are reporting exists but the conclusion is not entirely correct.

Given that I didn't quite understand the behavior and the proposed solution was based on said partial understanding, I'll close this issue for now if noone minds it.  (Simply reopen it if you do.) While the documentation could be more verbose about this, I'm not confident I would know what to add at this moment which is neither misleading nor inaccurate. Thanks both of you for helping me figure this out.
